src/main.cpp

```diff
@@ -879,14 +882,6 @@ bool ContextualCheckTransaction(
 
     // Rules that apply to Sapling and later:
     if (saplingActive) {
-        // Reject transactions with non-Sapling version group ID
-        if (tx.nVersionGroupId != SAPLING_VERSION_GROUP_ID) {
```

Comment:
I'm uncomfortable with applying Sapling-specific consensus checks to `tx.nVersion` without having first checked `tx.nVersionGroupId` (and likewise for other transaction versions).